### DISCUSSION TITLE
UCMM Unconnected Messaging

### DISCUSSION THREAD

**Discussion by Teedest** [EIPStackGroup/OpENer#472] [Q&A]
I have been playing with the sample application and have established Class 1 explicit communications with my end device. But I was wondering if there is support for creating/handling UCMM tags in OpENer?

**Comment by MartinMelikMerkumians** [comment]
Hi @Teedest 
not sure what you mean, but OpENer has UCMM support. As you closed your discussion, I guess it worked for you in the end?

**Reply by Teedest** [reply]
Thanks for the response. Yes, still a bit of a noob here and mixed up some of the terminology. What I was referring to was the way with Allen Bradley PLCs you can request the Logix tags directly to get data out of the PLC. The DCS I work with refers to this as "UCMM with Logix Tags" which confused me. My goal is to have a simple adapter we can use in house for testing comms mappings without the need for a PLC. I assume implementing something like this would be possible but you would have to create a custom service to handle the requests?

**Reply by MartinMelikMerkumians** [reply]
As far as I undertood EtherNet/IP, yes you would need a custom service, I think you would even need a custom object, as I don't know of any standard object providing this functionality.

However, I am not sure if OpENer would be then the thing you need. It's primarily designed to be an ENIP adapter device, so a fieldbus device controlled by a PLC. The PLC would then assign tags to the values acquired from the device to give it a name which can be used in addition to the corresponding EPath

**Reply by Teedest** [reply]
Yes normally it would be a PLC requesting data from a device but I'm working with a DCS(DeltaV) as the scanner and the Allen Bradley PLC is the adapter. The goal being to use OpENer to replace the need for a physical PLC if only to spit out static values to verify the DCS mappings/configuration for in house/FAT testing. For example I want to request the array "DCS_REAL" from the PLC(in this case an OpENer adapter) which contains 100 float/real values and show these in my configuration/displays. All the adapter would have to do is fill this array with values 0-100 so I can easily tell if I've mapped the data correctly or not.

**Reply by MartinMelikMerkumians** [reply]
Hmm, ok for that you would probably have either to know what object you need and how it works. You can get the class code most likely by analyzing the traffic between your DCS and the PLC. If it is a standard object class, you can probably easily implement it with a ENIP specification at hand.

If not, you probably need to reverse engineer the object and services based on the captured traffic

**Reply by Teedest** [reply]
> If not, you probably need to reverse engineer the object and services based on the captured traffic

Yeah that's the path I'm heading down for now. Thanks for the feedback. I'll update with how I get on and anything noteworthy I figure out.
